### PR TITLE
Fixes to minichart when updating data, disables legend events for minichart series

### PR DESCRIPTION
**Pull request by jacobp100**
https://ag-grid.atlassian.net/browse/AG-10675
https://ag-grid.atlassian.net/browse/AG-10653

### CODE REVIEW

File: packages/ag-charts-community/src/chart/chart.ts [hunk 1]
```diff
@@ -1647,14 +1659,30 @@ export abstract class Chart extends Observable implements AgChartInstance {
...
         }
 
+        const oldGrouping = target.seriesGrouping;
         if (seriesGrouping) {
             target.seriesGrouping = { ...target.seriesGrouping, ...(seriesGrouping as SeriesGrouping) };
         }
+
+        // This probably shouldn't be in this part of the flow
+        // However, the chart won't have access to the previous series grouping value to do this
+        if (applyLayers) {
```
Comment (jacobp100) [inline packages/ag-charts-community/src/chart/chart.ts]: I feel this definitely shouldn't exist here - but there's no way for a series to communicate to a chart that its grouping has changed. Maybe something like this:-

```ts
interface SeriesDelegate<T> {
    seriesGroupingDidChange: (series: Series<T>, oldValue?: SeriesGrouping) => void;
}

class Series<T> {
    delegate: SeriesDelegate?

      @ActionOnSet<Series<TDatum, TLabel>>({
        changeValue: function (newVal, oldVal) {
            this.onSeriesGroupingChange(oldVal, newVal);
            this.delegate?.seriesGroupingDidChange(this, oldVal)
        },
    })
    seriesGrouping?: SeriesGrouping = undefined;
}
```

Then the chart just has to implement `SeriesDelegate` and assign itself as the delegate to each series

Thoughts?

Comment (iMoses) [reply packages/ag-charts-community/src/chart/chart.ts]: From conversations I has with Alan it seems that we are heading towards using a centralized events-hub, which we'd like to use consistently as a mean of communication between modules as it promotes decoupling and provides easy access points to react to in future development.
So I'd go with managing it using events...

File: packages/ag-charts-community/src/chart/chart.ts [hunk 2]
```diff
@@ -1405,10 +1405,13 @@ export abstract class Chart extends Observable implements AgChartInstance {
...
         let seriesStatus: SeriesChangeType = 'no-op';
         if (processedOptions.series && processedOptions.series.length > 0) {
-            seriesStatus = this.applySeries(this, processedOptions);
+            seriesStatus = this.applySeries(this, processedOptions, oldOpts, true);
             forceNodeDataRefresh = true;
         }
-        if (this.applyAxes(this, completeOptions, seriesStatus)) {
+        if (seriesStatus === 'replaced') {
+            this.resetAnimations();
+        }
```
Comment (alantreadway) [inline packages/ag-charts-community/src/chart/chart.ts:1435]: We can't do this as it impacts Integrated Charts use-cases.

File: packages/ag-charts-community/src/chart/chart.ts (hunk 1 above)
Comment (alantreadway) [reply packages/ag-charts-community/src/chart/chart.ts]: Let sync on the need for this change, it's quite high risk and impacts correctness of stacking, grouping and Integrated Charts use cases?

File: packages/ag-charts-community/src/chart/chart.ts (hunk 2 above)
Comment (alantreadway) [reply packages/ag-charts-community/src/chart/chart.ts:1435]: Discussed offline: Moved from elsewhere, so this is a non-issue.

**Review by alantreadway — APPROVED** [review]
LGTM

**Pull request merged**